apps/src/studio/studio.js

```diff
@@ -1269,6 +1280,9 @@ studioApp.reset = function(first) {
     finishClipRect.setAttribute('x', Studio.spriteGoals_[i].x);
     finishClipRect.setAttribute('y', Studio.spriteGoals_[i].y);
   }
+
+  // A little flag for script-based code to consume.
+  window.studioLevelRestarted = true;
```

Comment:
Josh suggested using variables attached to window (in our infinity playlab chat room last week), for doing more advanced behaviour in the success javascript that's run every frame.  Here's an example from one of the levels:


function () {

  var currentTime = new Date().getTime();

  // Let's wait 2 seconds before indicating the success.
  if (window.successTime)
  {
    return successTime + 2000 < currentTime;
  }
  
  if (window.studioLevelRestarted || ! window.actorHit) {
    window.actorHit = [false, false, false, false];
  }
  
  window.studioLevelRestarted = false;
  
  var heroSprite = Studio.sprite[2];
  
  var hitCount = 0;
  
  var otherIndexes = [0,1,3,4];
  var heroIndex = 2;
  
  for (var i = 0; i < 4; i++) {
    
    actorIndex = otherIndexes[i];
    
    if (! window.actorHit[actorIndex] && Studio.sprite[actorIndex].isCollidingWith(heroIndex)) {
      Studio.sprite[actorIndex].dir = 1;
      window.actorHit[actorIndex] = true;
  	}
    
    if (window.actorHit[actorIndex]) {
      hitCount ++;
    }
  }
  
  var success = hitCount >= 2;
  
  if (! window.successTime && success)
  {
    //console.log("setting success");
    window.successTime = currentTime;
  }
}